common/djangoapps/student/models.py

```diff
@@ -1455,6 +1461,23 @@ def update_enrollment(self, mode=None, is_active=None, skip_refund=False):
                 self.send_signal(EnrollStatusChange.unenroll)
 
         if mode_changed:
+            if COURSEWARE_PROCTORING_IMPROVEMENTS.is_enabled(self.course_id):
```

Comment:
This block contain a lot of business logic that shouldn't belong in the models.py.
Maybe move this into [helpers.py](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/helpers.py) would make the code easier to read? This way, our models don't need to depend on classes like `IDVerificationService`